ytcc/database.py

```diff
@@ -157,25 +158,30 @@ def increase_extractor_fail_count(self, e_hash, max_fail=(1 << 63) - 1) -> None:
                 SET failure_count = failure_count + 1
                 WHERE failure_count < :max_fail
         """
-        self.connection.execute(query, {"e_hash": e_hash, "max_fail": max_fail})
+        with self.connection:
+            self.connection.execute(query, {"e_hash": e_hash, "max_fail": max_fail})
 
     def close(self) -> None:
         self.connection.commit()
         self.connection.close()
 
     def add_playlist(self, name: str, url: str) -> None:
         query = "INSERT INTO playlist (name, url) VALUES (?, ?);"
-        self.connection.execute(query, (name, url))
+        with self.connection:
+            res = self.connection.execute(query, (name, url))
+            return res.rowcount > 0
 
     def delete_playlist(self, name: str) -> bool:
         query = "DELETE FROM playlist WHERE name = ?"
-        res = self.connection.execute(query, (name,))
+        with self.connection:
+            res = self.connection.execute(query, (name,))
         return res.rowcount > 0
 
     def rename_playlist(self, oldname, newname) -> bool:
         query = "UPDATE playlist SET name = ? WHERE name = ?"
         try:
-            res = self.connection.execute(query, (newname, oldname))
+            with self.connection:
+                res = self.connection.execute(query, (newname, oldname))
```

Comment:
Same comment as above. Maybe move the return statement into the context?

Reply:
Agreed.